More drawing geometry: Point, Line, Box

dandelany opened the pull request:
This PR adds 3 new types of map geometry - Point, Line, and Box - which can be drawn by the user or added to the map programmatically with `addGeometry()`. A few notes:

* Line is just a LineString where the user is limited to drawing 2 points
* Box is a rectangle with sides that are north/south/east/west
* Cesium has the notion of an "extent" with north/south/east/west bounds; however, I decided to implement Box internally as a polygon instead, for simplicity and consistency with the other geometries which use vertex `coordinates`.

**ALSO:** While implementing this, I realized that the `addGeometry` function for Circles was inconsistent between the Cesium and OpenLayers map wrappers: the OpenLayers wrapper interpreted the `radius` value as being in meters, while the Cesium wrapper interpreted it as degrees of longitude. I discussed this with @AaronPlave  and we decided meters made more sense, so I've changed the OpenLayers wrapper to use meters. Not sure if this would be considered a breaking change for someone using the API or just a bugfix?

flynnplatt commented:
@dandelany it looks like the `Point` and `Box` draw types are missing the little dot under the cursor when drawing.

Also, the icon for the line segment could be clearer. Maybe the `vector-line` icon here: https://materialdesignicons.com/

dandelany commented:
No worries, thanks @flynnplatt - I've pushed the following updates to this branch:

* Updated `Point` and `Box` draw types to show the cursor dot while drawing
* Updated the menu icon for `Line` to use the `vector-line` icon you recommended above
* Updated the drawing tooltip text for the new draw types, so that it's relevant to each

Ready for another look!
